dist/gmusic.js

```diff
@@ -239,6 +245,87 @@ proto.playback = {
+    } else {
+      var evObj = document.createEvent('Events');
+      evObj.initEvent(etype, true, false);
+      el.dispatchEvent(evObj);
+    }
+  },
+
+  _render: function (container, force) {
+    // DEV: The queue isn't rendered till a click event is fired on this element
+    //      We must hide the queue during the 400ms animation and then reveal it
+    //      once both the 400ms in and 400ms out animations are complete
+    var table = container.querySelector('.queue-song-table');
+    if (container.style.display === 'none' && (!table || force)) {
+      // DEV: Hide the queue elements while we rapidly "render" the queue element
+      //      We have to use a style element because inline styles are saved by GPM
+      var style = document.createElement('style');
+      style.innerHTML = SELECTORS.queue.container + '{left: 10000px !important}';
+      document.body.appendChild(style);
+
+      // Render queue
+      this.queue._fireEvent(document.querySelector(SELECTORS.queue.triggerButton), 'click');
+      setTimeout(function () {
+        // Return queue to intitial state
+        this.queue._fireEvent(document.querySelector(SELECTORS.queue.triggerButton), 'click');
+        setTimeout(function () {
+          document.body.removeChild(style);
+        }, 20);
+      }.bind(this), 20);
+    }
+  },
+
+  clear: function (cb) {
+    var clearButton = this.doc.querySelector(SELECTORS.queue.container + ' ' + SELECTORS.queue.clearButton);
+    if (clearButton) {
+      clearButton.click();
+      setTimeout(function reRenderQueue () {
+        this.queue._render(this.doc.querySelector(SELECTORS.queue.container), true);
+        if (cb) {
+          cb();
+        }
+      }.bind(this), 200);
+    }
+  },
+
+  getSongs: function () {
+    var container = this.doc.querySelector(SELECTORS.queue.container);
+    this.queue._render(container);
+
+    return Array.prototype.slice.call(
+      container.querySelectorAll(SELECTORS.queue.songs)
+    ).map(function mapRowToSong (row) {
+      var timeString = row.querySelector('[data-col="duration"]').textContent.trim().split(':');
+      var details = row.querySelector('.song-details-wrapper');
+      var defaultString = {
+        textContent: null
+      };
+      var songObject = {
+        id: row.getAttribute('data-id'),
+        title: (details.querySelector('.song-title') || defaultString).textContent,
+        artist: (details.querySelector('.song-artist') || defaultString).textContent,
+        album: (details.querySelector('.song-album') || defaultString).textContent,
+        art: row.querySelector('[data-col="song-details"] img').src.replace('=s60-e100-c', ''),
+        duration: 1000 * (parseInt(timeString[0], 10) * 60 + parseInt(timeString[1], 10)),
+        playing: row.classList.contains('currently-playing')
+      };
+      return songObject;
+    });
+  },
+
+  playSong: function (id) {
+    var songRow = this.doc.querySelector('[data-id="' + id + '"]');
```

Comment:
`id` looks like this `"Ttdm3gyikzfmzuqxrlmzy2gy56q"` but I'm not sure how someone could pull off XSS in a `querySelector` call.

Calling `escape()` on the `id` would get rid of quotes and such.  Would that suffice for your anti-xss crusade :laughing: